Mage/src/main/java/mage/target/targetadjustment/ConditionalTargetAdjuster.java

```diff
@@ -4,48 +4,85 @@
 import mage.abilities.condition.Condition;
 import mage.game.Game;
 import mage.target.Target;
-import mage.target.Targets;
 
 /**
  * @author notgreat
  */
 public class ConditionalTargetAdjuster implements TargetAdjuster {
     private final Condition condition;
-    private final boolean keepExistingTargets;
-    private final Targets replacementTargets;
+    private final boolean keepBlueprintTarget;
+    private final Target replacementTarget;
+    private Target blueprintTarget;
```

Comment:
I find these variable names and constructors quite confusing tbh. Why "blueprintTarget"?

Reply:
`blueprintTarget` is because it fulfills fundamentally the same role as the `blueprintTarget` in `GenericTargetAdjuster` (but since it allows the null target, it isn't exactly identical).

Reply:
Alright, I looked at the updated code instead of the diff, and it's definitely documented properly.